x/concentrated-liquidity/swaps_test.go

```diff
@@ -0,0 +1,162 @@
+				expectedTokenOut:  sdk.NewInt(66808387),
+			},
+		},
+		{
+			name: "out is lesser than min amount",
+			param: param{
+				tokenIn:           sdk.NewCoin("foo", sdk.NewInt(42000000)),
+				tokenOutDenom:     "bar",
+				tokenOutMinAmount: sdk.NewInt(8397),
+			},
+			expectErr: true,
+		},
+		{
+			name: "in and out denom are same",
+			param: param{
+				tokenIn:           sdk.NewCoin("bar", sdk.NewInt(13370)),
+				tokenOutDenom:     "bar",
+				tokenOutMinAmount: sdk.NewInt(1),
+			},
+			expectErr: true,
+		},
+		{
+			name: "unknown in denom",
+			param: param{
+				tokenIn:           sdk.NewCoin("bara", sdk.NewInt(13370)),
+				tokenOutDenom:     "bar",
+				tokenOutMinAmount: sdk.NewInt(1),
+			},
+			expectErr: true,
+		},
+		{
+			name: "unknown out denom",
+			param: param{
+				tokenIn:           sdk.NewCoin("bar", sdk.NewInt(13370)),
+				tokenOutDenom:     "bara",
+				tokenOutMinAmount: sdk.NewInt(1),
+			},
+			expectErr: true,
+		},
+	}
+
+	for _, test := range tests {
+		s.Run(test.name, func() {
+			// Init suite for each test.
+			s.Setup()
+
+			// Create a foo - bar concentrated liquidity pool
+			pool, err := s.App.ConcentratedLiquidityKeeper.CreateNewConcentratedLiquidityPool(s.Ctx, 1, "bar", "foo", currSqrtPrice, currTick)
+			s.Require().NoError(err)
+
+			// Check the test case to see if we are swapping asset0 for asset1 or vice versa
+			asset0 := pool.GetToken0()
+			zeroForOne := test.param.tokenIn.Denom == asset0
+
+			// Fund the test account with foo and bar, then create a default position to the pool created earlier
+			s.FundAcc(s.TestAccs[0], sdk.NewCoins(sdk.NewCoin("bar", sdk.NewInt(10000000000000)), sdk.NewCoin("foo", sdk.NewInt(1000000000000))))
+			_, _, _, err = s.App.ConcentratedLiquidityKeeper.CreatePosition(s.Ctx, 1, s.TestAccs[0], defaultAmt0, defaultAmt1, sdk.ZeroInt(), sdk.ZeroInt(), lowerTick.Int64(), upperTick.Int64())
+			s.Require().NoError(err)
+
+			if test.expectErr {
+				pool, err = s.App.ConcentratedLiquidityKeeper.GetPoolById(s.Ctx, pool.GetId())
+				_, err = s.App.ConcentratedLiquidityKeeper.SwapExactAmountIn(s.Ctx, s.TestAccs[0], pool.(swaproutertypes.PoolI), test.param.tokenIn, test.param.tokenOutDenom, test.param.tokenOutMinAmount, swapFee)
+				s.Require().Error(err)
+			} else {
+				spotPriceBefore := pool.GetCurrentSqrtPrice().Power(2)
+				prevGasConsumed := s.Ctx.GasMeter().GasConsumed()
+
+				// Execute the swap directed in the test case
+				tokenOutAmount, err := s.App.ConcentratedLiquidityKeeper.SwapExactAmountIn(s.Ctx, s.TestAccs[0], pool.(swaproutertypes.PoolI), test.param.tokenIn, test.param.tokenOutDenom, test.param.tokenOutMinAmount, swapFee)
+				s.Require().NoError(err)
+				s.Require().Equal(test.param.expectedTokenOut.String(), tokenOutAmount.String())
+
+				gasConsumedForSwap := s.Ctx.GasMeter().GasConsumed() - prevGasConsumed
+				// TODO: make a CLGasFeeForSwap
+				// Check that we consume enough gas that a CL pool swap warrants
```

Comment:
nit: seems like a quick change. Do you mind adding it right away instead of having to track this?